packages/core/src/browser/about.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2018 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+import { injectable, inject } from 'inversify';
+import { Message } from '@phosphor/messaging';
+import { AbstractDialog } from './dialogs';
+import { MenuContribution, MenuModelRegistry } from '../common/menu';
+import { Command, CommandContribution, CommandRegistry } from '../common/command';
+import { IApplicationServer, ExtensionInfo } from '../common/application-protocol';
+import { CommonMenus } from './common-frontend-contribution';
+
+export const ABOUT_CONTENT_CLASS = 'theia-aboutDialog';
+export const ABOUT_EXTENSIONS_CLASS = 'theia-aboutExtensions';
+
+export class AboutDialog extends AbstractDialog<void> {
+
+    protected readonly okButton: HTMLButtonElement;
+
+    constructor(extensionsInfos: ExtensionInfo[]) {
+        super({
+            title: `About Theia`
+        });
+
+        const messageNode = document.createElement('div');
+        messageNode.classList.add(ABOUT_CONTENT_CLASS);
+
+        const extensionInfoTitle = document.createElement('h2');
+        extensionInfoTitle.textContent = 'Extensions';
+        messageNode.appendChild(extensionInfoTitle);
+
+        const extensionInfoContent = document.createElement('ul');
+        extensionInfoContent.classList.add(ABOUT_EXTENSIONS_CLASS);
+        messageNode.appendChild(extensionInfoContent);
+
+        extensionsInfos.forEach(extension => {
+            const extensionInfo = document.createElement('li');
+            extensionInfo.textContent = extension.name + ' ' + extension.version;
+            extensionInfoContent.appendChild(extensionInfo);
+        });
+        messageNode.setAttribute('style', 'flex: 1 100%; padding-bottom: calc(var(--theia-ui-padding)*3);');
+        this.contentNode.appendChild(messageNode);
+        this.appendAcceptButton('Ok');
+    }
+
+    protected onAfterAttach(msg: Message): void {
+        super.onAfterAttach(msg);
+    }
+
+    get value(): undefined { return undefined; }
+}
+
+export const ABOUT_COMMAND: Command = {
+    id: 'core.about',
+    label: 'About'
+};
+
+@injectable()
+export class AboutContribution implements MenuContribution, CommandContribution {
+
+    @inject(IApplicationServer)
+    protected readonly appServer: IApplicationServer;
+
+    registerMenus(registry: MenuModelRegistry): void {
+
+        registry.registerMenuAction(CommonMenus.HELP, {
+            commandId: ABOUT_COMMAND.id,
+            label: 'About',
+            order: '9'
+        });
+    }
+
+    registerCommands(commandRegistry: CommandRegistry): void {
+        commandRegistry.registerCommand(ABOUT_COMMAND, {
+            execute: async () => {
```

Comment:
please extract method for closures that extenders can change the implementation, i.e. `openAbout`

Reply:
What's the best way to make this openAbout extensible for theia application developers? I would need an external dependency to CommonFrontendContribution that developers can bind that would provide an `openAbout` method is that what you mean?

Reply:
I meant to convert a closure to a method, so one can subclass and override it.